pkg/kubectl/cmd/apply/apply_test.go

```diff
@@ -279,12 +284,20 @@ func TestRunApplyPrintsValidObjectList(t *testing.T) {
 		NegotiatedSerializer: resource.UnstructuredPlusDefaultContentConfig().NegotiatedSerializer,
 		Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
 			switch p, m := req.URL.Path, req.Method; {
-			case strings.HasPrefix(p, pathCM) && m != "GET":
-				pod := ioutil.NopCloser(bytes.NewReader(cmBytes))
-				return &http.Response{StatusCode: 200, Header: cmdtesting.DefaultHeader(), Body: pod}, nil
-			case strings.HasPrefix(p, pathCM) && m != "PATCH":
-				pod := ioutil.NopCloser(bytes.NewReader(cmBytes))
-				return &http.Response{StatusCode: 200, Header: cmdtesting.DefaultHeader(), Body: pod}, nil
+			case strings.HasPrefix(p, pathCM) && m == "GET":
+				fallthrough
+			case strings.HasPrefix(p, pathCM) && m == "PATCH":
+				var body io.ReadCloser
+
+				// "test" is the first ConfigMap in the ConfigMapList
+				// "test2" is the second one
+				if strings.HasSuffix(p, "test") {
```

Comment:
Can we do better than that? The dependency on extremely implicit names is confusing.